### ISSUE TITLE
SslSession：Data lost if sending continuously

### ISSUE THREAD

**Issue by cuiliang** [chronoxor/NetCoreServer#57]
if send new packet before previous packet is sent, the new packet will not send to client.
```
// in TrySend()
if (_sending)
                    return;
```
perhaps there should be a buffer manager, each send need a separate buffer to store data to send.

**Comment by chronoxor** [comment]
If you look into `SendAsync()` you'll see that data stores in `_sendBufferMain` and `TrySend()/ProcessSend()` will work until all data is send.

**Comment by cuiliang** [comment]
I'v added some Debug.WriteLine to the code and got some information:
https://github.com/cuiliang/NetCoreServer/blob/master/source/NetCoreServer/SslSession.cs

Here's the output:
```
SendAsync(byte[] buffer) Sending Bytes:124
_sending is True
Writing to sslStream Bytes:124
SendAsync(byte[] buffer) Sending Bytes:54
Return Pos 4
Clearing _isSending
```
When sending the second 54 byte packet, _isSending is TRUE, so the call to TrySend just returned without writing to sslStream.

**Comment by cuiliang** [comment]
when the first packet is sending, the second packet was writed to  _sendBufferMain,  but after the first packet send, no TrySend was called to send data in _sendBufferMain.

```
// code in ProcessSend()
if (!result.CompletedSynchronously)
                {
                    TrySend();
                }
```
If add TrySend to  'else' seems can fix the problem, but I dont know if it's the right way to do it.

**Comment by cuiliang** [comment]
In the Remarks section of this document :
https://docs.microsoft.com/en-us/dotnet/api/system.iasyncresult.completedsynchronously?view=netframework-4.8#remarks
```
If the synchronous completion of the call is detected in the AsyncCallback delegate, it is probable that the thread that initiated the asynchronous operation is the current thread.
```
Perhaps that's the reason of the problem.

**Comment by chronoxor** [comment]
Yes some data might be send asynchronously with some small delay. I suggest you calculating how exact bytes were sent to the peer by overriding `protected virtual void OnSent(long sent, long pending) {}` method and count bytes there.

**Comment by cuiliang** [comment]
It's not delay, it will never send the second packet if I do not send the 3rd. 
I think it's a bug that should be fixed.

**Comment by chronoxor** [comment]
Ok I see. Could you please comment out the condition below in `ProcessSend()`:
```
                // Try to send again if the session is valid
                //if (!result.CompletedSynchronously)
                TrySend();
```

**Comment by chronoxor** [comment]
The same fix should be done in `ProcessReceive()`:
```
                // If zero is returned from a read operation, the remote end has closed the connection
                if (size > 0)
                {
                    //if (!result.CompletedSynchronously)
                    TryReceive();
                }
```

If this fix the issue, I'll commit the fix into master.

**Comment by cuiliang** [comment]
Yes, I'v tested, comment that line will work.  Thank you~

**Comment by chronoxor** [comment]
Fixed in NetCoreServer 3.0.11

**Comment by D3c0ded** [comment]
I have the same problem.
I'm using v6.7.0 of NetCoreServer.
